src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -138,17 +139,26 @@ public static RequestDelegateResult Create(Delegate handler, RequestDelegateFact
         var factoryContext = CreateFactoryContext(options, handler);
 
         Expression<Func<HttpContext, object?>> targetFactory = (httpContext) => handler.Target;
-
         var targetableRequestDelegate = CreateTargetableRequestDelegate(handler.Method, targetExpression, factoryContext, targetFactory);
 
-        if (targetableRequestDelegate is null)
+        RequestDelegate finalRequestDelegate = targetableRequestDelegate switch
         {
             // handler is a RequestDelegate that has not been modified by a filter. Short-circuit and return the original RequestDelegate back.
             // It's possible a filter factory has still modified the endpoint metadata though.
-            return new RequestDelegateResult((RequestDelegate)handler, AsReadOnlyList(factoryContext.Metadata));
+            null => (RequestDelegate)handler,
+            _ => httpContext => targetableRequestDelegate(handler.Target, httpContext),
+        };
+
+        if (targetableRequestDelegate is null)
```

Comment:
😄 couldn't decide whether to use a switch expression or if statement so went with both?